common/logger/init.go

```diff
@@ -0,0 +1,94 @@
+package logger
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+
+	"os"
+	"path/filepath"
+)
+
+const (
+	DefaultTimeFormat           = "15:04:05.000"
+	DefaultCallerSkipFrameCount = 4
+
+	UseColor  = true
+	UseCaller = false // if you want to expose line of code of caller
+	LogFile   = ""    // if you want to store log to file
+)
+
+var (
+	logBuffer bytes.Buffer
+
+	// DebugMode flag for determining debug mode
+	DebugMode = false
+)
+
+func init() {
+	var (
+		lgr zerolog.Logger
+	)
+
+	zerolog.TimeFieldFormat = DefaultTimeFormat
+	zerolog.CallerSkipFrameCount = DefaultCallerSkipFrameCount
+
+	var writers zerolog.LevelWriter
+	consoleWriter := zerolog.ConsoleWriter{
+		Out:        &logBuffer,
+		NoColor:    UseColor,
+		TimeFormat: DefaultTimeFormat,
+	}
+	writers = zerolog.MultiLevelWriter(consoleWriter)
+
+	file, err := openLogFile()
+	if err == nil && file != nil {
+		writers = zerolog.MultiLevelWriter(writers, file, consoleWriter)
+	}
+
+	lgr = zerolog.New(writers)
+
+	if UseCaller {
+		lgr = lgr.With().Caller().Logger()
```

Comment:
If all of the log methods are wrapped in logger.go, do the printed lines all end up resolving to the logger.go file?

Reply:
I've set `DefaultCallerSkipFrameCount` to 3, so it will print the filepath whos calling logger.go

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/YO1Dcg4NByYdZHvKXaTq/bed170af-922a-43a9-9937-e8d4cdd7bb83.png)

Reply:
👍